metronome/hotstuff/service/src/io/iohk/metronome/hotstuff/service/ConsensusService.scala

```diff
@@ -229,11 +233,40 @@ class ConsensusService[F[_]: Timer: Concurrent, N, A <: Agreement: Block](
       applySyncEffects(state, effects)
 
     // Unstash messages before we change state.
-    unstash(nextState) >>
+    captureChanges(nextState) >>
+      unstash(nextState) >>
```

Comment:
I'm a bit confused by the comment above. Does the order really matter here?

Reply:
Ahh, because unstash operates on the current `stateRef` :+1:

Reply:
Yeah, I don't know if I should have threaded the state itself to all these methods, seemed more awkward.